linker/shared/src/main/scala/org/scalajs/linker/frontend/optimizer/IncOptimizer.scala

```diff
@@ -851,6 +915,137 @@ final class IncOptimizer private[optimizer] (config: CommonPhaseConfig, collOps:
       }
     }
 
+    /** UPDATE PASS ONLY. */
+    private def interpretConstructor(impl: MethodImpl,
+        fieldBodies: Map[(ClassName, FieldName), FieldBody],
+        paramBodies: List[Option[FieldBody]]): Map[(ClassName, FieldName), FieldBody] = {
+
+      /* This method performs a kind of abstract intepretation of the given
+       * given constructor `impl`. It *assumes* that the enclosing class ends
+       * up having elidable constructors. If that is not the case, the
+       * computation must not crash but its result can be arbitrary.
+       */
+
+      type FieldBodyMap = Map[(ClassName, FieldName), FieldBody]
+      type ParamBodyMap = Map[LocalName, FieldBody]
+
+      def interpretSelfGetter(methodName: MethodName,
+          fieldBodies: FieldBodyMap): Option[FieldBody] = {
+        methods.get(methodName).flatMap { impl =>
+          impl.originalDef.body match {
+            case Some(Select(This(), className, FieldIdent(fieldName))) =>
+              fieldBodies.get((className, fieldName))
+
+            case _ =>
+              /* If we get here, it means the class won't have elidable
+               * constructors, and therefore we can return arbitrary values
+               * from the whole method, so we don't care.
+               */
+              None
+          }
+        }
+      }
+
+      def interpretExpr(tree: Tree, fieldBodies: FieldBodyMap,
+          paramBodies: ParamBodyMap): Option[FieldBody] = {
+        tree match {
+          case lit: Literal =>
+            Some(FieldBody.Literal(lit))
+
+          case VarRef(LocalIdent(valName)) =>
+            paramBodies.get(valName)
+
+          case LoadModule(moduleClassName) =>
+            Some(FieldBody.LoadModule(moduleClassName))
+
+          case Select(LoadModule(moduleClassName), className, FieldIdent(fieldName)) =>
+            if (moduleClassName == Class.this.className) {
```

Comment:
Do we still need this branch? IIUC if we get a self-load module now, we consider the class to have non-elidable ctors?

Reply:
Good catch. We don't need it anymore. Nor the other one.